Address issues discovered during RC integration testing

bgeltz opened the pull request:
- Relates to #3388.

bgeltz committed:
Use the app basename when specifying the filter

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
Fix test_hint_time.py

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
RawReportCollection: Handle reports from non-MPI apps

- Unmarked will not be present.

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
Fix test_time.py

- Run with GEOPM_CTL_LOCAL to avoid MPI_Init
  in geopmctl.
- Increase the app duration and timeout
  to account for slow systems.
- Increase the global time limit.

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

dannosliwcd commented on geopmpy/geopmpy/io.py:
```diff
@@ -890,13 +894,13 @@ def load_reports(self, reports, dir_name, dir_cache, verbose, do_cache):
-                        self._unmarked_reports_df.to_hdf(self._report_h5_name, 'unmarked_report', format='table', append=True)
-                        self._epoch_reports_df.to_hdf(self._report_h5_name, 'epoch_report', format='table', append=True)
+                        self._app_reports_df.to_hdf(self._report_h5_name, 'app_report', format='table')
+                        if len(self._reports_df) > 0:
+                            self._reports_df.to_hdf(self._report_h5_name, 'report', format='table', append=True)
+                        if len(self._unmarked_reports_df) > 0:
+                            self._unmarked_reports_df.to_hdf(self._report_h5_name, 'unmarked_report', format='table', append=True)
+                        if len(self._epoch_reports_df) > 0:
+                            self._epoch_reports_df.to_hdf(self._report_h5_name, 'epoch_report', format='table', append=True)
```
Why are the `if(len)` checks needed now (or were they needed all along)?

It seems like it has something to do with the fact that we initialize these through multiple maybe-executed lines because of a string-related issue.

bgeltz replied:
I was having trouble with test_time.py and running with --skip-launch.  The issue was that I was trying to query for the various dataframes with the getters for RawReportCollection, only in some cases I was getting an exception that they didn't exist, and in some cases I was getting back an empty dataframe.

It turns out it was all related to the code that tries to load the HDF5 cache of reports.  If it was the first time and the data was actually parsed and cached, it was all OK, but if the test was trying to read from the cache and do the same operation, it failed.

Let me double check on this today and ensure this is all in fact needed.

dannosliwcd commented on integration/test/geopm_test_launcher.py:
```diff
@@ -128,7 +129,7 @@ def run(self, test_name, include_geopm_policy=True, add_geopm_args=[], enable_af
                                                 '--geopm-agent', self._agent_conf.get_agent(),
                                                 '--geopm-profile', test_name,
-                                                '--geopm-program-filter', self._app_conf.get_exec_path()]
+                                                '--geopm-program-filter', os.path.basename(self._app_conf.get_exec_path())]
```
Should I apply something similar here as well? 
* Add a `app_conf.get_program_filter()` function, defaulting to exec path but allowing individual appconf implementations to override it https://github.com/geopm/geopm/pull/3062/files#diff-76dd7a1a9349bf0d8cf85212926694ed5006d12dcb6a8b64e04c04a4da7a42ea
* Use `get_program_filter()` in launch_util.py https://github.com/geopm/geopm/pull/3062/files#diff-343359e818df706352ab39f88c63ca674e14f518dcc4eacc1fbd88feea99520d

Or could we do something similar to those changes as part of this PR so it applies for all users of appconf?

bgeltz replied:
Chris and I discussed this and thought it would be a good idea to make the change in the launcher.py.  If I did it there, all these downstream usages would not have to change.  Let's discuss.

bgeltz committed:
Use only the basename for paths provided in GEOPM_PROGRAM_FILTER

- Required for job step launchers that copy the
  app binary to a non-standard location (e.g.
  mpiexec).

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

cmcantalupo commented on libgeopm/src/Environment.cpp:
```diff
@@ -382,10 +382,13 @@ namespace geopm
         if (is_set("GEOPM_PROGRAM_FILTER")) {
             auto valid_names = geopm::string_split(lookup("GEOPM_PROGRAM_FILTER"), ",");
+            // Extract just the basename for any path provided as input
+            std::transform(valid_names.begin(), valid_names.end(), valid_names.begin(),
+                           [](const std::string& vn) {
+                               return vn.substr(vn.find_last_of("/") + 1);});
```
Does this work when only the base name is provides as input?
